### DISCUSSION TITLE
Creating an OEM Module

### DISCUSSION THREAD

**Discussion by lukereed** [SunPower/PVMismatch#156] [Q&A]
I am trying to create a `pvmismatch.pvmismatch_lib.pvmodule.PVmodule` object that is reflective of Trina TSM-PD14 module. I am having difficulty creating a module that is the correct number of cells (72) and the correct wattage (315W). Whenever I make a 72 cell module, the corresponding PV Mismatch Module is ~240W peak power. Is there an easy way to set up a module with custom power?

**Comment by mikofski** [comment]
If you have the data sheet from Trina with Isc, Vmp, Imp, and Voc try using https://sunpower.github.io/PVMismatch/api/contrib.html#module-pvmismatch.contrib.gen_coeffs to get the diode model coefficients

by default the diode model coefficients set in https://sunpower.github.io/PVMismatch/api/pvcell.html#id2 are for the SunPower SPR-E20-315 which has 96 cells, that’s why you get a 240W module because 72/96*315W = 236.25W

I found this data sheet: https://static.trinasolar.com/sites/default/files/EN_TSM_PD14_datasheet_B_2017_web.pdf

I don’t see a 315W but the 320W has
- Vmp = 37.1V
- Imp = 8.63A
- Voc = 45.5V
- Isc = 9.15A


You’ll also need `alpha_isc` which is on the data sheet too, it’s 0.05%/K off Isc

**Reply by lukereed** [reply]
Awesome, thanks! So that provides me with the `isat1`, `isat2`, `rs`, and `rsh` coefficients, but it appears that the [`PVcell`](https://sunpower.github.io/PVMismatch/api/pvcell.html#id2) object also requires the `Isco_TO`, `aRBD`, `bRBD`, `VRBD`, and `nRBD` coefficients as well. Are these able to be calculated, or do they not change from module to module?

**Reply by mikofski** [reply]
`Isc0_T0` is the short circuit current at the reference temperature, 25°C, already provided by Trina in the data sheet, I think it was 9.15 amps.

The other parameters are breakdown coefficients, just leave them at default values, because chances are you will never go into reverse bias or activate any bypass diode

**Reply by lukereed** [reply]
Excellent, thank you very much!

**Comment by mikofski** [comment]
Just out of curiosity what is your use case? Are you also planning to use pvlib? Are you going to present or publish anywhere?

**Reply by lukereed** [reply]
We (RES) are trying to quantify the mismatch losses behind each inverter block at existing operational sites where we know the string composition. This is to determine the value associated with the PV module mismatch and degradation at a particular plant. I was planning to use the [Annual Mismatch Loss Calculator](https://github.com/SunPower/MismatchLossStudy/blob/master/run_annual_energy_mismatch.py) to derive this loss, but may end up doing more detailed analysis by using pvlib later.

As far as presenting and publishing goes, it depends a bit on what we find when modeling our first few sites. If we find some value potential in it, then we will certainly at least present on it.

**Reply by mikofski** [reply]
Let me know if you think there anything that DNV could help with. My email address is on my profile page. I hope you find PVMismatch useful, thanks for trying it, and best of luck!

**Reply by lukereed** [reply]
Will do, and thanks again for your help!
